src/amplitude/http_client.py

```diff
@@ -134,4 +134,14 @@ def post(url: str, payload: bytes, header=None) -> Response:
         except timeout:
             result.code = 408
             result.status = HttpStatus.TIMEOUT
+        except error.HTTPError as e:
+            try:
+                result.parse(e)
+            except:
+                result = Response()
+                result.code = e.code
+                result.status = Response.get_status(e.code)
+                result.body = dict(error=e.msg)
+        except error.URLError as e:
+            result.body = dict(error=e.reason)
```

Comment:
URLError.reason can be str or another exception instance. Better convert it to str here.
```suggestion
            result.body = dict(error=str(e.reason))
```